contract/src/main.rs

```diff
@@ -1560,110 +1561,116 @@ pub extern "C" fn set_token_metadata() {
                     NFTCoreError::InvalidTotalTokenSupply,
-                )
-            {
-                runtime::revert(NFTCoreError::CannotUpgradeToMoreSupply)
+                );
+                storage::write(total_token_supply_uref, total_token_supply_arg);
+                total_token_supply_arg
             }
-
-            let total_token_supply_uref = utils::get_uref(
-                ARG_TOTAL_TOKEN_SUPPLY,
+            None => utils::get_stored_value_with_user_errors::<u64>(
+                TOTAL_TOKEN_SUPPLY,
                 NFTCoreError::MissingTotalTokenSupply,
                 NFTCoreError::InvalidTotalTokenSupply,
-            );
-            storage::write(total_token_supply_uref, total_token_supply_arg);
-            total_token_supply_arg
+            ),
+        };
+
+        if total_token_supply == 0 {
+            runtime::revert(NFTCoreError::CannotUpgradeWithZeroSupply)
         }
-        None => utils::get_stored_value_with_user_errors::<u64>(
-            TOTAL_TOKEN_SUPPLY,
-            NFTCoreError::MissingTotalTokenSupply,
-            NFTCoreError::InvalidTotalTokenSupply,
-        ),
-    };
 
-    if total_token_supply == 0 {
-        runtime::revert(NFTCoreError::CannotUpgradeWithZeroSupply)
-    }
+        let current_number_of_minted_tokens = utils::get_stored_value_with_user_errors::<u64>(
+            NUMBER_OF_MINTED_TOKENS,
+            NFTCoreError::MissingNumberOfMintedTokens,
+            NFTCoreError::InvalidNumberOfMintedTokens,
+        );
 
-    let current_number_of_minted_tokens = utils::get_stored_value_with_user_errors::<u64>(
-        NUMBER_OF_MINTED_TOKENS,
-        NFTCoreError::MissingNumberOfMintedTokens,
-        NFTCoreError::InvalidNumberOfMintedTokens,
-    );
+        if total_token_supply < current_number_of_minted_tokens {
+            runtime::revert(NFTCoreError::ExceededMaxTotalSupply)
+        }
 
-    if total_token_supply < current_number_of_minted_tokens {
-        runtime::revert(NFTCoreError::ExceededMaxTotalSupply)
-    }
+        let requires_rlo_migration = utils::requires_rlo_migration();
 
-    if requires_rlo_migration {
-        storage::new_dictionary(PAGE_TABLE)
-            .unwrap_or_revert_with(NFTCoreError::FailedToCreateDictionary);
-        let page_table_width = utils::max_number_of_pages(total_token_supply);
-        runtime::put_key(PAGE_LIMIT, storage::new_uref(page_table_width).into());
-        runtime::put_key(
-            REPORTING_MODE,
-            storage::new_uref(OwnerReverseLookupMode::Complete as u8).into(),
-        );
+        if !requires_rlo_migration && runtime::get_key(RLO_MFLAG).is_some() {
+            return;
+        }
 
-        let collection_name = utils::get_stored_value_with_user_errors::<String>(
-            COLLECTION_NAME,
-            NFTCoreError::MissingCollectionName,
-            NFTCoreError::InvalidCollectionName,
-        );
+        if requires_rlo_migration {
```

Comment:
```suggestion
       ! if requires_rlo_migration {
            return;
            }
       
```

Reply:
Exit early here if possible

Reply:
> Exit early here if possible

It does not seem to compile anymore

Reply:
Basically with the new spec the condition can be 
```
        if !utils::requires_rlo_migration() || runtime::get_key(RLO_MFLAG).is_some() {
            return;
        }
```